main.go

```diff
@@ -111,6 +112,8 @@ func main() {
 		os.Exit(1)
 	}
 
+	reconciler.CreateCRIfNotPresent()
```

Comment:
I'd simplify this with a GET call on the resource, if not found, then call the create function.